api/src/test/java/uk/gov/dhsc/htbhf/claimant/entity/ClaimTest.java

```diff
@@ -63,6 +60,16 @@ void shouldFailToValidateClaimWithoutClaimant() {
         assertThat(violations).hasSingleConstraintViolation("must not be null", "claimant");
     }
 
+    @Test
+    void shouldFailToValidateClaimWithInvalidEligibilityOverride() {
```

Comment:
Can we make this a parameterised test or more tests with more permutations of the two fields in `EligibilityOverride` please?

Reply:
below is coming in next PR with parameterised
```
@ParameterizedTest
    @MethodSource("eligibilityOverrideWithNullValues")
    void shouldFailToValidateEligibilityOverrideWithNullValues(EligibilityOverride eligibilityOverride) {
        boolean result = eligibilityOverrideValidator.isValid(eligibilityOverride, null);

        assertThat(result).isFalse();
    }

    private static Stream<EligibilityOverride> eligibilityOverrideWithNullValues() {
        return Stream.of(
                EligibilityOverride.builder()
                        .eligibilityOutcome(null)
                        .overrideUntil(OVERRIDE_UNTIL_FIVE_YEARS)
                        .childrenDob(emptyList())
                        .build(),
                EligibilityOverride.builder()
                        .eligibilityOutcome(EligibilityOutcome.CONFIRMED)
                        .overrideUntil(null)
                        .childrenDob(emptyList())
                        .build(),
                EligibilityOverride.builder()
                        .eligibilityOutcome(EligibilityOutcome.CONFIRMED)
                        .overrideUntil(OVERRIDE_UNTIL_FIVE_YEARS)
                        .childrenDob(null)
                        .build(),
                EligibilityOverride.builder()
                        .eligibilityOutcome(null)
                        .overrideUntil(null)
                        .childrenDob(null)
                        .build()
        );
    } 
```

Reply:
I still think it would be worthwhile adding this to this PR, you can then simply add in the childrenDob values in the next PR.